File: interpreter/interpreter/ast.py
from .token import Token

class Node:
    pass

class Number(Node):
    def __init__(self, token: Token):
        self.token = token

    def __str__(self):
        return f"Number ({self.token})"

class BinOp(Node):
    def __init__(self, left: Node, op: Token, right: Node):
        self.left = left
        self.op = op
        self.right = right

    def __str__(self):
        return f"BinOp{self.op.value} ({self.left}, {self.right})"

class UnarOp(Node):
    def __init__(self, op: Token, node: Node):
        self.node = node
        self.op = op

    def __str__(self):
        return f"UnOp{self.op.value} {self.node}"

File: pascalInterpreter/src/main/kotlin/interpreter/Ast.kt
package interpreter

abstract class Node(){
}

data class Number(val token: Token) : Node()
data class BinOp(val left: Node, val op: Token, val right: Node) : Node()
data class UnarOp(val op: Token, val value: Node) : Node()
data class Empty(val value: String = "") : Node()
data class Variable(val token: Token, val value: Node) : Node()
data class Identifier(val token: Token) : Node()
data class Semicolon(val left: Node, val right: Node) : Node()
